pysal/contrib/viz/plot.py

```diff
@@ -9,38 +9,44 @@
 import pysal as ps
 import matplotlib.pyplot as plt
 
-
-def moran(var, w, xlabel='', ylabel='', title='', custom=(5,5)):
-	'''
+def mplot(m, xlabel='', ylabel='', title='', custom=(7,7)):
+    '''
     Produce basic Moran Plot 
     ...
-    Arguments
+    Parameters
     ---------
-    var	            : array
-                      values of variable
-    w       	    : spatial weight
+    m            : array
```

Comment:
oops, it is: m.__class__ = <class pysal.esda.moran.Moran at 0x10d995a10>

Though while I was looking into this, and type(m) after getting an m, I got
class "instance," which after some reading is considered an old-style class
of type classobj,rather than inheriting from the object (see this
<http://stackoverflow.com/questions/6666856/why-does-typemyfield-return-type-instance-and-not-type-field>
discussion and this background
<http://www.cafepy.com/article/python_types_and_objects/python_types_and_objects.html>).
Is there a better way to inherit the right class for m in the applied side,
or is that a function of m?

On Sun, May 22, 2016 at 3:54 PM, Dani Arribas-Bel <notifications@github.com>
wrote:

> In pysal/contrib/viz/plot.py
> <https://github.com/pysal/pysal/pull/797#discussion_r64156092>:
>
> >      ---------
> > -    var	            : array
> > -                      values of variable
> > -    w       	    : spatial weight
> > +    m            : array
>
> It's not an array, it's a ps.Moran object, isn't it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/pysal/pysal/pull/797/files/beace6676894fa61fd6a1328b8b0cc08e1fd8b49..6da47d169b5d14cf5d35592c41601653f4cc11b7#r64156092>
>